### DISCUSSION TITLE
Is it possible to plot volume candle chart?

### DISCUSSION THREAD

**Discussion by githubloki** [highfestiva/finplot#416] [General]
Hi, I want to plot a candlestick chart whose candle width are based on volume. I tried to pass df.width into candle_width argument but gives an error citing series error.

I.e :  fplt.candlestick_ochl (df, candle_width=df.width, ax=ax)

Error : --------
TypeError: arguments did not match any overloaded call:
  QRectF(): too many arguments
  QRectF(QPointF, QSizeF): argument 1 has unexpected type 'Series'
  QRectF(QPointF, QPointF): argument 1 has unexpected type 'Series'
  QRectF(float, float, float, float): argument 1 has unexpected type 'Series'
  QRectF(QRect): argument 1 has unexpected type 'Series'
  QRectF(QRectF): argument 1 has unexpected type 'Series'

**Comment by highfestiva** [comment]
The width parameter is not variable across a seris, but constant. I recommend you use a value between 0.1 (10% of width between candles) and 1 (100% of space between candles used).

**Reply by githubloki** [reply]
Thanks for replying,
I have adjusted values of df.width within 0 to 1. # 0< df.width <1

max = df.Volume.max()
df['width']= df.Volume / max
fplt.candlestick_ochl(df['Open Close High Low width'.split()], ax=ax)

a little change in sourcecode
line 1155 df,origlen = self.datasrc.rows(6, left, right, yscale=self.ax.vb.yscale)
line 1164 for x,(t,open,close,high,low, width) in zip(idxs, rows):
line 1170 for x,(t,open,close,high,low, width) in zip(idxs, rows):
line 1171 p.drawRect(QtCore.QRectF(x-width/2, open, width, close-open))

This plots near to my expectation but not exactly same as I desired because this leave huge gap between candles.

This is what i want.
[https://school.stockcharts.com/doku.php?id=chart_analysis:candlevolume](https://github.com/highfestiva/finplot/discussions/url)

**Comment by highfestiva** [comment]
You'll have to keep track of your "last x+width", which always is the "next x". You can't just iterate over it, but you're close.

**Reply by githubloki** [reply]
Tried but failed, maybe I should modify axis tick in pyqtgraph but I'm not a pro coder.
Not sure how to do this.
